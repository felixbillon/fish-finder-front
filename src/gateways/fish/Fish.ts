export interface Fish {
  id?: number;
  commonName: string;
  scientificName: string;
  maxLenght: number;
  maxWeight: number;
}
