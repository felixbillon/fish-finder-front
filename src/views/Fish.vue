<template>
  <div class="fish">
    <FishList />
  </div>
</template>

<script lang="ts">
import FishList from "@/components/FishList.vue";
import { defineComponent } from "@vue/composition-api";

export default defineComponent({
  name: "fish",
  components: {
    FishList
  }
});
</script>
