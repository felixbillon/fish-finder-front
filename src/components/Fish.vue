<template>
  <div>
    {{ fish.commonName }}
  </div>
</template>

<script lang="ts">
import { createComponent } from "@vue/composition-api";
import { Fish } from "@/gateways/fish/Fish";

interface Props {
  fish: Fish;
}

export default createComponent<Props>({
  name: "Fish",
  props: {
    fish: Object
  }
});
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss"></style>
