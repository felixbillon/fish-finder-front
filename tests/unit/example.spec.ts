import { mount } from "@vue/test-utils";
import Fish from "@/components/Fish.vue";

describe("Fish.vue", () => {
  it("should have a custom title and match snapshot", () => {
    const wrapper = mount(Fish, {
      propsData: {
        fish: {
          commonName: "salomon"
        }
      }
    });

    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot();
  });
});
