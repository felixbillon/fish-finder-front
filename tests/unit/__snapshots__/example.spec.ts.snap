// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fish.vue should have a custom title and match snapshot 1`] = `
<div class="v-card v-sheet theme--light">
  <div tabindex="-1" class="v-list-item v-list-item--three-line theme--light">
    <div class="v-list-item__content">
      <div>FISH</div>
      <div class="v-list-item__title">salomon</div>
      <div class="v-list-item__subtitle">Test</div>
    </div>
    <div class="v-avatar v-list-item__avatar v-avatar--tile grey" style="height: 80px; min-width: 80px; width: 80px;"></div>
  </div>
</div>
`;
